lib/edools/lesson_progress.rb

```diff
@@ -0,0 +1,148 @@
+# frozen_string_literal: true
+
+module Edools
+  class LessonProgress
+    attr_reader :lesson_progress, :paginate, :errors
+
+    def initialize(data = {})
+      @lesson_progress = objectify_lesson_progress(data)
+      @errors = data.dig(:errors)
+    end
+
+    def self.find_paginated_from_enrollments(enrollment_id: , page: 1, per_page: 50)
+      raise ArgumentError, 'missing enrollment_id' if enrollment_id.nil?
+
+      url = "#{base_url_enrollments}/#{enrollment_id}/lessons_progresses"
+      data = { enrollment_id: enrollment_id, page: page, per_page: per_page }
+
+      response = Edools::ApiRequest.request(:get, url, data)
+
+      OpenStruct.new(
```

Comment:
Maybe we could return something like a "collectionProxy" here , allowing the user a more straightforward approach like `Edools::LessonProgresses.find_paginated_from_enrollments.first` ? I think that we can ship it with your current structure as it is easy to keep retro-compatibility and add an issue for providing CollectionProxy-like capabilities in the future.

Reply:
the collection looks like this:
`Edools::LessonProgresses.find_paginated_from_enrollments.lessons_progresses.first`

Reply:
Exactly , that behavior is not what one would normally expect hence the request to make clean the return definition inside the README, it's not common the need to refer to what object you want right after doing a find on the class that define that object.

A CollectionProxy-like would allow us to retain the expected behaviour (i.e. `Edools::LessonProgresses.find_paginated_from_enrollments.first` ) and at the same time to refer to pagination related information.